backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/S3Service.java

```diff
@@ -30,18 +33,37 @@
   private static final Logger LOGGER = LogManager.getLogger(S3Service.class);
   private AmazonS3 s3Client;
   private String bucketName;
+  private String keyName;
+  private String UploadFileName;
 
   public S3Service(String cloudBucketName) {
     App app = App.getInstance();
     String cloudAccessKey = app.getCloudAccessKey();
     String cloudSecretKey = app.getCloudSecretKey();
-    if (cloudAccessKey != null && cloudSecretKey != null) {
+    String minioEndpoint = app.getMinioEndpoint();
+    if (cloudAccessKey != null && cloudSecretKey != null && minioEndpoint == null) {
```

Comment:
I think this will improve the readability

```
if (cloudAccessKey != null && cloudSecretKey != null){
 if(minioEndpoint ==null){
 ...
 } else {
  //your new code 
 }
} else { 
  //read from env
}
```

Reply:
Agreed. I've adapted the file to this structure and I've also removed a couple of variables which were not used 
```
-  private String keyName;
-  private String UploadFileName;
```